src/components/AvgProductsPaymentWidget/AvgProductsPaymentWidget.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Crm\ProductsModule\Components;
+
+use Crm\ApplicationModule\Widget\BaseWidget;
+use Crm\ApplicationModule\Widget\WidgetManager;
+use Crm\UsersModule\Repository\UserMetaRepository;
+
+class AvgProductsPaymentWidget extends BaseWidget
+{
+    private $templateName = 'avg_products_payment_widget.latte';
+
+    private $userMetaRepository;
+
+    public function __construct(WidgetManager $widgetManager, UserMetaRepository $userMetaRepository)
+    {
+        parent::__construct($widgetManager);
+        $this->userMetaRepository = $userMetaRepository;
+    }
+
+    public function identifier()
+    {
+        return 'avgproductspaymentwidget';
+    }
+
+    public function render(array $userIds)
+    {
+        if (count($userIds)) {
```

Comment:
_(general advise, no need to change this)_

If there's nothing to render, you can stop the method immediately:

```php
if (!count($userIds)) {
    return;
}

// sunny day scenario processing

if (!$average->avg_product_payment) {
   // no need to render the view if you can stop here right away
    return;
}
```

It's easier to read: you can see when the processing stops, it's not indented within extra IF branch and you can "follow" the sunny day scenario.